src/main/java/com/thealgorithms/dynamicprogramming/SumOfSubset.java

```diff
@@ -1,18 +1,37 @@
 package com.thealgorithms.dynamicprogramming;
 
-public class SumOfSubset {
+public final class SumOfSubset {
 
-    public static boolean subsetSum(int[] arr, int num, int Key) {
-        if (Key == 0) {
+    private SumOfSubset() {
+    }
+
+    public static boolean subsetSum(int[] arr, int n, int key) {
```

Comment:
After applying the suggestion about the _outer loop_, the `n` seems to be unused. Can one simplify the signature of this function and remove `n`?

Reply:
But what about test cases , i write the test case where the methods takes signature like(arr,lastindex,key)?

Reply:
Please update them.

Notice that the original implementation is recursive _over_ the `lastindex`.